packages/NodeTypeResolver/src/NodeTypeResolver.php

```diff
@@ -795,4 +840,46 @@ private function getClassNodeProperty(ClassLike $classLike, string $name): ?Prop
 
         return null;
     }
+
+    private function getVendorPropertyFetchType(PropertyFetch $propertyFetch): ?Type
+    {
+        $varObjectType = $this->getStaticType($propertyFetch->var);
+        if (! $varObjectType instanceof TypeWithClassName) {
+            return null;
+        }
+
+        if ($this->parsedNodesByType->findClass($varObjectType->getClassName())) {
+            return null;
+        }
+
+        // 3rd party code
+        $propertyName = $this->nameResolver->getName($propertyFetch->name);
+        if ($propertyName === null) {
+            return null;
+        }
+
+        if (! property_exists($varObjectType->getClassName(), $propertyName)) {
+            return null;
+        }
+
+        // property is used
+        $propertyReflection = new ReflectionProperty($varObjectType->getClassName(), $propertyName);
+        if (! $propertyReflection->getDocComment()) {
+            return null;
+        }
+
+        $phpDocNode = $this->betterPhpDocParser->parseString((string) $propertyReflection->getDocComment());
+        $varTagValues = $phpDocNode->getVarTagValues();
+
+        if (! isset($varTagValues[0])) {
+            return null;
+        }
+
+        $typeNode = $varTagValues[0]->type;
+        if (! $typeNode instanceof TypeNode) {
+            return null;
+        }
+
+        return $this->staticTypeMapper->mapPHPStanPhpDocTypeNodeToPHPStanType($typeNode, new Nop());
+    }
```

Comment:
@gnutix It's not perfect solution. But it's the only one without parsing the vendor file (which is performance and missing-classes trap).

Here we deduce type from the doc of property, thanks to php reflection.